lib/src/mt_ptp.c

```diff
@@ -431,7 +481,9 @@ static int ptp_sync_expect_result(struct mt_ptp_impl* ptp) {
       ptp_calculate_coefficient(ptp, ptp->expect_result_avg);
     }
   }
+#ifndef MTL_HAS_DPDK_TIMESYNC_ADJUST_FREQ
```

Comment:
Use a fake delta to adjust clock, will break the LOCKED status of pi_sample() call.

Reply:
This will break the error case for E810. Can we add one argument to ptp_adjust_delta API like below?
```
static void ptp_adjust_delta(struct mt_ptp_impl* ptp, int64_t delta, bool error_correct) {
  /* only run phc2sys if not error_correct */
  if (!error_correct && mt_has_phc2sys_service(ptp->impl)) phc2sys_adjust();
}
```

Reply:
ptp: distinguish the adjustment because of real delta or average delta correction.

    To ice device, real delta correction wasn't enough for expected precision. Average
    delta correction was added for additional adjustment.

    To igc device, real delta correction was enough. But the average delta correction will
    violate the precision being established by real delta correction.

    Considering ice device not enabling phc2sys service originally, while the phc2sys
    service isn't requested, execute the both real delta and average delta correction
    like original adjustment procedure. So, the ice device keeps the original feature
    and precision.

    When the phc2sys service is requested, only execute the real delta correction. It
    make the time sync based on igc device achieves best precision.

> This will break the error case for E810. Can we add one argument to ptp_adjust_delta API like below?
> 
> ```
> static void ptp_adjust_delta(struct mt_ptp_impl* ptp, int64_t delta, bool error_correct) {
>   /* only run phc2sys if not error_correct */
>   if (!error_correct && mt_has_phc2sys_service(ptp->impl)) phc2sys_adjust();
> }
> ```